rules/automount-service-account/raw.rego

```diff
@@ -27,170 +30,184 @@ deny [msga]{
-	fix_path = ""
+    # automountServiceAccountToken set to true in pod spec
+    spec.automountServiceAccountToken == true
+    
+    # No SA (yaml scan)
+    service_accounts := [service_account | service_account = input[_]; service_account.kind == "ServiceAccount"]
+    count(service_accounts) == 0
+    failed_path = sprintf("%vautomountServiceAccountToken", [beggining_of_path])
+    fix_path = ""
 }
 
 
 
  #  -- ----     For SAs     -- ----     
 is_auto_mount(service_account)  =  [failed_path, fix_path]  {
-	service_account.automountServiceAccountToken == true
-	failed_path = "automountServiceAccountToken"
-	fix_path = ""
+    service_account.automountServiceAccountToken == true
+    failed_path = "automountServiceAccountToken"
+    fix_path = ""
 }
 
 is_auto_mount(service_account)=  [failed_path, fix_path]  {
-	not service_account.automountServiceAccountToken == false
-	not service_account.automountServiceAccountToken == true
-	fix_path = {"path": "automountServiceAccountToken", "value": "false"}
-	failed_path = ""
+    not service_account.automountServiceAccountToken == false
+    not service_account.automountServiceAccountToken == true
+    fix_path = {"path": "automountServiceAccountToken", "value": "false"}
+    failed_path = ""
 }
 
 is_same_sa(spec, serviceAccountName) {
-	spec.serviceAccountName == serviceAccountName
+    spec.serviceAccountName == serviceAccountName
 }
 
 is_same_sa(spec, serviceAccountName) {
-	not spec.serviceAccountName 
-	serviceAccountName == "default"
+    not spec.serviceAccountName 
+    serviceAccountName == "default"
 }
 
 
 is_same_namespace(metadata1, metadata2) {
-	metadata1.namespace == metadata2.namespace
+    metadata1.namespace == metadata2.namespace
 }
 
 is_same_namespace(metadata1, metadata2) {
-	not metadata1.namespace
-	not metadata2.namespace
+    not metadata1.namespace
+    not metadata2.namespace
 }
 
 is_same_namespace(metadata1, metadata2) {
-	not metadata2.namespace
-	metadata1.namespace == "default"
+    not metadata2.namespace
+    metadata1.namespace == "default"
 }
 
 is_same_namespace(metadata1, metadata2) {
-	not metadata1.namespace
-	metadata2.namespace == "default"
-}
\ No newline at end of file
+    not metadata1.namespace
+    metadata2.namespace == "default"
+}
+
+has_service_account_binding(serviceAccount) {
+    role_bindings := [role_binding | role_binding = input[_]; role_binding.kind == "RoleBinding"]
+    role_binding := role_bindings[_]
+    role_binding.subjects[_].name == serviceAccount.metadata.name
+    role_binding.subjects[_].kind == "ServiceAccount"
+}
+
+has_service_account_binding(service_account) {
+    cluster_role_bindings := [cluster_role_binding | cluster_role_binding = input[_]; cluster_role_binding.kind == "ClusterRoleBinding"]
+    cluster_role_binding := cluster_role_bindings[_]
+    cluster_role_binding.subjects[_].name == service_account.metadata.name
+    cluster_role_binding.subjects[_].kind == "ServiceAccount"
+}
```

Comment:
We need to add a check for namespace, there can be many `default` service accounts, for e.g